### PR TITLE
Rebump JNA with LIBFFI_TMPDIR

### PR DESCRIPTION
**Pull request by wi11dey**
This reverts commit 5816ffc83084137fd11e9a24be75005861fcb53c.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -416,7 +416,7 @@
             <exclusion groupId="hsqldb" artifactId="hsqldb"/> <!-- outdated version -->
             <exclusion groupId="antlr" artifactId="antlr"/> <!-- outdated version -->
           </dependency>
-          <dependency groupId="net.java.dev.jna" artifactId="jna" version="4.0.0"/>
+          <dependency groupId="net.java.dev.jna" artifactId="jna" version="5.8.0"/>
```
Comment (schlosna) [inline build.xml]: This needs to be at least 5.10.0 to pick up the fix https://github.com/java-native-access/jna/pull/1378

```suggestion
          <dependency groupId="net.java.dev.jna" artifactId="jna" version="5.13.0"/>
```

**Review by schlosna — CHANGES_REQUESTED** [review]
We should bump JNA to at least 5.10.0, although latest is 5.13.0, see https://github.com/java-native-access/jna/blob/master/CHANGES.md

File: bin/cassandra.in.sh [hunk 2]
```diff
@@ -99,6 +99,7 @@ JAVA_OPTS="$JAVA_OPTS:-Djava.library.path=$CASSANDRA_HOME/lib/sigar-bin"
 CASSANDRA_TEMP_DIR=$SERVICE_HOME/var/data/tmp
 JVM_OPTS="${JVM_OPTS} -Djava.io.tmpdir=$CASSANDRA_TEMP_DIR"
 JVM_OPTS="${JVM_OPTS} -Djna.tmpdir=$CASSANDRA_TEMP_DIR"
+export LIBFFI_TMPDIR="$CASSANDRA_TEMP_DIR"
```
Comment (schlosna) [inline bin/cassandra.in.sh:104]: We may also want to set and export `TMPDIR` for any other native libraries that do tricky things:

```suggestion
# See https://access.redhat.com/solutions/6991416
# https://davecturner.github.io/2021/08/30/seven-year-old-segfault.html
export LIBFFI_TMPDIR="$CASSANDRA_TEMP_DIR"
export TMPDIR="$CASSANDRA_TEMP_DIR"
```

File: build.xml (hunk 1 above)
Comment (rhuffy) [reply build.xml]: I would support using 5.13, as that's what's used on Cassandra trunk https://github.com/apache/cassandra/blob/830f88d4ab2aaa1f7b88b90747a808d0a4e2a708/.build/parent-pom-template.xml#L548

**Comment by wi11dey** [comment]
SHA256 of jna-5.13.0.jar: `66d4f819a062a51a1d5627bffc23fac55d1677f0e0a1feba144aabdd670a64bb`
SHA1 of jna-5.13.0.jar from maven: `1200e7ebeedbe0d10062093f32925a912020e747`
@rhuffy

**Comment by schlosna** [comment]
> SHA1 of jna-5.13.0.jar from maven: `1200e7ebeedbe0d10062093f32925a912020e747`

Validated https://github.com/palantir/cassandra/raw/3c64de216ed4079d5ce2be8936599aa278a7439c/lib/jna-5.13.0.jar against https://repo1.maven.org/maven2/net/java/dev/jna/jna/5.13.0/jna-5.13.0.jar.sha1 and locally

```
$ shasum -a 1 jna-5.13.0.jar
1200e7ebeedbe0d10062093f32925a912020e747  jna-5.13.0.jar
```

**Review by schlosna — APPROVED** [review]

**Comment by wi11dey** [comment]
Tested Cassandra is booting up healthy even when tmpfs is mounted noexec

**Review by zpear — APPROVED** [review]

**Pull request merged**